cmd/root.go

```diff
@@ -47,6 +42,7 @@ func Execute() error {
 // command including all flags and sub-commands.
 // This function is principally used for tests.
 func BuildRootCommand() *cobra.Command {
+	cfg := &config{}
```

Comment:
Making config scoped to a given `BuildRootCommand` invocation, rather than global scoped, will reduce possible footguns when building E2E tests in this codebase - each test case can call `BuildRootCommand` to get a fresh config.